src/toast/noise_sim.py

```diff
@@ -138,93 +138,209 @@ def _detector_weight(self, det):
         wt = 1.0 / (self._NET[det] ** 2) / self._rate[det]
         return wt.decompose()
 
-    def _save_hdf5(self, handle, comm, **kwargs):
-        """Internal method which can be overridden by derived classes."""
-        # First save the base class info
-        self._save_base_hdf5(handle, comm)
-
+    def _gather_sim(self, comm, out):
+        # Gather simulation properties to the rank zero process
+        if comm is None or comm.size == 1:
+            out["fmin"] = self._fmin
+            out["fknee"] = self._fknee
+            out["rate"] = self._rate
+            out["alpha"] = self._alpha
+            out["NET"] = self._NET
+            return out
+
+        pfmin = comm.gather(self._fmin, root=0)
+        pfknee = comm.gather(self._fknee, root=0)
+        prate = comm.gather(self._rate, root=0)
+        palpha = comm.gather(self._alpha, root=0)
+        pnet = comm.gather(self._NET, root=0)
+        if comm.rank == 0:
+            out["fmin"] = dict()
+            out["fknee"] = dict()
+            out["rate"] = dict()
+            out["alpha"] = dict()
+            out["NET"] = dict()
+            for pr, pfm, pfk, pa, pn in zip(prate, pfmin, pfknee, palpha, pnet):
+                out["rate"].update(pr)
+                out["fmin"].update(pfm)
+                out["fknee"].update(pfk)
+                out["alpha"].update(pa)
+                out["NET"].update(pn)
```

Comment:
This function returns None, and modifies the `out` dictionary in place.  The error is actually on line 150 where I returned this rather than just having an empty return.